src/plugins/navigation_embeddable/public/components/editor/navigation_embeddable_panel_editor.tsx

```diff
@@ -124,13 +124,16 @@ const NavigationEmbeddablePanelEditor = ({
           setOrderedLinks(
             orderedLinks.map((link) => {
               if (link.id === linkToEdit.id) {
-                return { ...newLink, order: linkToEdit.order };
+                return { ...newLink, order: linkToEdit.order } as NavigationEmbeddableLink;
               }
               return link;
             })
           );
         } else {
-          setOrderedLinks([...orderedLinks, { ...newLink, order: orderedLinks.length }]);
+          setOrderedLinks([
+            ...orderedLinks,
+            { ...newLink, order: orderedLinks.length } as NavigationEmbeddableLink,
+          ]);
```

Comment:
I was getting some local Typescript errors here that I think I missed in my linking PR when I was trying to wrap it up quickly - so this should fix it 😅